File: solutions/1154/main.py
from typing import List 

class Solution:
    def __init__(self):
        self.months = {}
        for i in [1, 3, 5, 7, 8, 10, 12]:
            self.months[i] = 31
        for i in [4, 6, 9, 11]:
            self.months[i] = 30
        self.months[2] = 28
    @staticmethod
    def checkLeap(year):
        if year % 100 == 0 and year % 400 == 0:
            return True
        if year % 4 == 0:
            return True
        return False
    def dayOfYear(self, date: str) -> int:
        result = 0
        y, m, d = date.split("-")
        y, m, d = int(y), int(m), int(d)
        leap = self.checkLeap(y)
        if leap:
            self.months[2] += 1
        for i in range(1, m):
            result += self.months[i]
        result += d
        return result

if __name__ == "__main__": 
    date = "2019-01-09"
    s = Solution()
    assert s.dayOfYear(date) == 9

File: solutions/1154/main.cpp
/* ************************************************************************
> File Name:     main.cpp
> Author:        Sean
> Project:    LeetCode
> Created Time:  Mon Dec 20 09:11:24 2021
> Description:   
 ************************************************************************/
#include <vector> 
#include <gtest/gtest.h> 
#include <string> 

using namespace std; 

class Solution {
private:
    vector<int> months;
public:
    Solution() : months(12, 31) {
        months[1] = 28;
        months[3] = 30;
        months[5] = 30;
        months[8] = 30;
        months[10] = 30;
    }
    bool checkLeap(int year) {
        if (year % 100 == 0 && year % 4 == 0) return true;
        if (year % 4 == 0) return true;
        return false;
    }
    int dayOfYear(string date) {
        int y = stoi(date.substr(0, 4));
        int m = stoi(date.substr(5, 2));
        int d = stoi(date.substr(8, 2));
        if (checkLeap(y)) months[1]++;
        int result = 0;
        for (int i = 0; i < m-1; ++i) {
            result += months[i];
        }
        result += d;
        return result;
    }
};

TEST(Test1154, SimpleTest) {
    string date = "2019-01-09";
    Solution s; 
    EXPECT_EQ(s.dayOfYear(date), 9);
}

int main() {
    testing::InitGoogleTest();
    return RUN_ALL_TESTS();
}
